libs/entry-components/configure-theming.md

```diff
@@ -0,0 +1,491 @@
+
+## Styles Configuration
+
+In summary, the generator:
+
+- Includes multiple generator files for supported entry components (dialog, table, button, form...)
+- Generates [Angular Material](https://material.angular.io/guide/theming) for customization of color palettes, typography, and density settings
+- Sets up  default, boilerplate theme with predefined values
+- Combines default theme and custom theme styles
+
+```scss
+@use 'entry-components/styles/generator';
+```
+
+## Create custom theme
+
+Next, define the $custom-theme map that will be expanded with various customization options based on specific needs.
+
+```scss
+$custom-theme: ();
+```
+
+Include and apply the styles generated by the generate-from mixin in the 'generator' module, using the customization options provided in the $custom-theme map.
+
+```scss
+@include generator.generate-from($custom-theme);
+```
+
+Our theme is now an empty object, and default library theme is used.
+
+```scss
+@use 'entry-components/styles/generator';
+
+$custom-theme: (
+);
+
+@include generator.generate-from($custom-theme);
+
+```
+
+## Importing Fonts
+
+There are 2 possibilities of using fonts in theming:
+
+### 1. Use default Roboto font
+
+**Entry-components** library comes with the default **Roboto** font family. To utilize it, configuration in the `angular.json` file at the root of the app is necessary. The font is imported from the library's node modules as follows:  
+
+```json
+"assets": [
+  "src/favicon.ico",
+  "src/assets",
+  {
+    "glob": "**/*",
+    "input": "./node_modules/@enigmatry/entry-components/assets/",
+    "output": "/assets/"
+  }
+],
+```
+
+### 2. Preparing custom fonts
+
+Ensure that the following steps are completed:
+
+1. Place custom font files in the `assets/fonts/` folder of project.
+2. Create a dedicated SCSS file (e.g., `partials/core/fonts.scss`)
+3. Use the `@font-face` rule in `fonts.scss` to define custom fonts with an absolute path.
+
+```scss
+@font-face {
+  font: {
+    family: 'Helvetica';
+    weight: 400;
+  }
+  src: url('/assets/fonts/Helvetica.woff2') format('woff2'), url('/assets/fonts/Helvetica.woff') format('woff');
+}
+```
+
+### Configuration Properties
+
+The new **custom theme map** should replace the library's default theme. The structure is nested and defines various aspects of a theme. It can be extended and customized for project needs by passing on different key parameters. The structure consists of three main sections: **general**, **tables**, and **dialogs**. Each section contains a specific configuration.
+
+> **_NOTE:_**  Any property in the theming configuration can be skipped, and if omitted, the **default value** for that property will be used (unless it's specified differently, as for fonts). This provides flexibility for developers to customize only the specific aspects of the theme that are relevant to their project. See table with default values [here](#default-values-table)
```

Comment:
Please don't forget to link "as for fonts" to new paragraph (once done) where you'll explain what happens to fonts if you define only some of them.

Reply:
I've included a new paragraph related to that under the "Fonts and Prerequisites" section with a corresponding link. Some parts of that section will be moved to the documentation that is specifically for setting up from scratch.

Reply:
We'll deal with generic things in other ticket, but we need to explain HERE why avoiding some property in theming map is totally fine EXCEPT for i.e. fonts. That has nothing to do with setup from scratch.